pkg/apis/security/v1alpha1/httppolicy_validation.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2020 Google LLC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"net/http"
+
+	"knative.dev/pkg/apis"
+)
+
+var validHTTPMethods = map[string]bool{
+	http.MethodConnect: true,
+	http.MethodDelete:  true,
+	http.MethodGet:     true,
+	http.MethodHead:    true,
+	http.MethodOptions: true,
+	http.MethodPatch:   true,
+	http.MethodPost:    true,
+	http.MethodPut:     true,
+	http.MethodTrace:   true,
+}
+
+// Validate validates a HTTPPolicy.
+func (p *HTTPPolicy) Validate(ctx context.Context) *apis.FieldError {
+	var errs *apis.FieldError
+	if p.Spec.JWT != nil {
+		if err := p.Spec.JWT.Validate(ctx); err != nil {
+			errs = errs.Also(err.ViaField("jwt").ViaField("spec"))
+		}
+	}
+	if p.Spec.Rules != nil {
```

Comment:
Nit: here and other similar places not necessary to check for nil...  range of a nil slice just returns no matches.  I'm ok with it as-is though.

Reply:
Done